test/e2e/Helper.sol

```diff
@@ -0,0 +1,230 @@
+        return SettlementEncoder.EncodedSettlement({
+            tokens: new IERC20[](0),
+            clearingPrices: new uint256[](0),
+            trades: new GPv2Trade.Data[](0),
+            interactions: [new GPv2Interaction.Data[](0), new GPv2Interaction.Data[](0), new GPv2Interaction.Data[](0)]
+        });
+    }
+
+    function _deployBalancerVaultAt(address _vault) internal {
+        bytes memory authorizerInitCode = abi.encodePacked(_getBalancerBytecode("Authorizer"), abi.encode(owner));
+        address authorizer = _create(authorizerInitCode, 0);
+
+        bytes memory vaultInitCode = abi.encodePacked(_getBalancerBytecode("Vault"), abi.encode(authorizer, WETH, 0, 0));
+        vm.record();
+        address deployedVault = _create(vaultInitCode, 0);
+        (, bytes32[] memory writeSlots) = vm.accesses(deployedVault);
+
+        // replay storage writes made in the constructor and set the balancer code
+        vm.etch(address(_vault), deployedVault.code);
+        for (uint256 i = 0; i < writeSlots.length; i++) {
+            bytes32 slot = writeSlots[i];
+            bytes32 val = vm.load(deployedVault, slot);
+            vm.store(address(_vault), slot, val);
+        }
+
+        // grant required roles
+        vm.startPrank(owner);
+        IAuthorizer(authorizer).grantRole(
+            _getActionId("manageUserBalance((uint8,address,uint256,address,address)[])", address(deployedVault)),
+            vaultRelayer
+        );
+        IAuthorizer(authorizer).grantRole(
+            _getActionId(
+                "batchSwap(uint8,(bytes32,uint256,uint256,uint256,bytes)[],address[],(address,bool,address,bool),int256[],uint256)",
+                address(deployedVault)
+            ),
+            vaultRelayer
+        );
+        vm.stopPrank();
+    }
+
+    function _getActionId(string memory fnDef, address vaultAddr) internal pure returns (bytes32) {
+        bytes32 hash = keccak256(bytes(fnDef));
+        bytes4 selector = bytes4(hash);
+        return keccak256(abi.encodePacked(uint256(uint160(vaultAddr)), selector));
+    }
+
+    function _getBalancerBytecode(string memory artifactName) internal view returns (bytes memory) {
+        string memory data = vm.readFile(string(abi.encodePacked("balancer/", artifactName, ".json")));
+        return vm.parseJsonBytes(data, ".bytecode");
+    }
+
+    function _create(bytes memory initCode, uint256 value) internal returns (address deployed) {
+        assembly ("memory-safe") {
+            deployed := create(value, add(initCode, 0x20), mload(initCode))
+        }
+    }
+
+    function deployMintableErc20(string memory name, string memory symbol) internal returns (IERC20Mintable token) {
+        // need to use like this because OZ requires ^0.7 and tests are on ^0.8
+        bytes memory initCode =
+            abi.encodePacked(vm.getCode("test/e2e/ERC20Mintable.sol:ERC20Mintable"), abi.encode(name, symbol));
```

Comment:
Blocker here https://github.com/foundry-rs/foundry/issues/8842

Reply:
https://github.com/cowprotocol/contracts/pull/215/commits/5894ee1d9637ed6c895c0c837ba0ac26441ffaeb this is a workaround, since `test/src/` is already softlinked into `src/contracts/test` and that makes the file available for all tests. it is the only way to get it to be able to find a file that is not imported in the test file to be available via `getCode`. see https://github.com/foundry-rs/foundry/issues/8842#issuecomment-2342897191